src/main/java/uk/gov/ons/fwmt/job_service/rest/impl/UserResourceServiceImpl.java

```diff
@@ -36,28 +36,50 @@ public UserResourceServiceImpl(
   public Optional<UserDto> findByAuthNo(String authNo) {
     log.info("findByAuthNo: {}", authNo);
     try {
-      final ResponseEntity<UserDto> responseEntity = restTemplate.getForEntity(findUrl, UserDto.class, authNo);
-      if (responseEntity.getStatusCode().equals(HttpStatus.OK)) {
-        return Optional.ofNullable(responseEntity.getBody());
+      final ResponseEntity<UserDto> userDto = restTemplate.getForEntity(findUrl, UserDto.class, authNo);
+      if (userDto.getStatusCode().equals(HttpStatus.OK)) {
+        return Optional.ofNullable(userDto.getBody());
+      } else {
+        // any success that doesn't have a 200 is an unexpected error
+        log.warn("findByAuthNo: returned with a non-200 code: authNo={}, code={}", authNo, userDto.getStatusCodeValue());
+        return Optional.empty();
       }
     } catch (HttpClientErrorException httpException) {
-      log.error("An error occurred while communicating with the resource service", httpException);
+      if (httpException.getStatusCode() != HttpStatus.NOT_FOUND) {
+        // any other unexpected error
+        log.warn(String.format("findByAuthNo: error communicating with the resource service: authNo=%s", authNo), httpException);
+        return Optional.empty();
+      } else {
+        // a 404, which occurs when we can't find an authNo
+        log.info("findByAuthNo: authNo not found", httpException);
+        return Optional.empty();
+      }
     }
```

Comment:
Personally I find the style below easier to read. Interesting discussion here: https://stackoverflow.com/q/2928556/

```java
} catch (HttpClientErrorException httpException) {
    log.error("An error occurred while communicating with the resource service", httpException);
    // a 404, which occurs when we can't find an authNo
    if (httpException.getStatusCode() == HttpStatus.NOT_FOUND) {
        log.info("findByAuthNo: authNo not found", httpException);
        return Optional.empty();
    }

    // any other unexpected error
    log.warn(String.format("findByAuthNo: error communicating with the resource service: authNo=%s", authNo), 
        httpException);
    return Optional.empty();
}
```